.github/workflows/rust.yml

```diff
@@ -47,7 +47,13 @@ jobs:
     - run: rustup update
     - run: rustup target add ${{ matrix.target }}
     - run: cargo clippy --all-targets --target ${{ matrix.target }}
-    - run: cargo test --target ${{ matrix.target }}
+    # unit tests
+    - run: cargo test --bins --target ${{ matrix.target }}
+    # integration tests
+    # TODO: we should probably manage the container in the test itself
+    # TODO: do a proper functional check on the container startup instead of waiting 15 seconds
+    - run: docker run -d -p 8000:8000 stellar/quickstart:soroban-dev --standalone --enable-soroban-rpc --protocol-version 20 && sleep 15
```

Comment:
While this will work today, adding a dependency on stellar-core to the CLI tests will probably prevent many PRs from being merged on this repo. That's because stellar-core is rarely updated and released with the new version of the env interface, but the env interface is under active development, so any time you update the env without also getting the core team to build a new deb, the tests dependent on the quickstart will break.

I imagine we need to iterate on this test setup, so could we move the quickstart tests into a separate job that won't block PRs from merging?

Reply:
We've been running integration tests on Horizon (which depend on Core) for more than a year without many problems. They break once in a while but it hasn't been a problem IMO.

Regarding the `env` changes, bear in mind this is a very basic test. Do you expect the env to break deploying and invoking a basic contract?